basics/uri.adoc

```diff
@@ -140,3 +323,50 @@ NOTE: Local URIs do not contain the authority (and begin with `/`)
 
 
 NOTE: Please also refer to https://devhints.io/semver[devhints.io/semver] for more examples of version strings
+
+
+== Best Practices
+
+A UUri contain UAuthority, UEntity, and UResource. <<Uri Formats>> defined earlier merely define how UAuthority, UEntity, and UResource are expressed (long string names, short-form string, or encoded byte array). Some formats are human readable, while others are not.
+
+uProtocol URI specifications complies with of https://datatracker.ietf.org/doc/html/rfc3986[RFC3986] that builds https://datatracker.ietf.org/doc/html/rfc1034[DNS concepts and facilities] to allow representation of UAuthorty as names or IP addresses. 
+
+UEntity name to id, and UResource name+instance+message to id, are declared in the protos and compiled into language specific code using  https://protobuf.dev/programming-guides/proto2/#options[Options]. Application layer (uP-L3) uEs (client and servers) then have available the all the metadata needed to populate UUri with both the ids and names for UEntity and UResource as well the application layer code can also convert to/from the name-based (long Uri Format) to ID based (Short or Micro Uri Formats) locally (no dynamic lookup required). uEs are then able to send to the link:../up-l1/README.adoc[uP-L1 uTransport layer] The fully populated UUri objects.
```

Comment:
I think this is a typo "then have available the all the metadata needed"
I think there is a "the" too many.